openedx/core/djangoapps/catalog/utils.py

```diff
@@ -114,38 +120,46 @@ def munge_catalog_program(catalog_program):
     }
 
 
-def get_course_run(course_key, user):
-    """Get a course run's data from the course catalog service.
+def get_course_runs(user, course_keys=[]):  # pylint: disable=dangerous-default-value
+    """
+    Get a course run's data from the course catalog service.
 
     Arguments:
-        course_key (CourseKey): Course key object identifying the run whose data we want.
+        course_keys (CourseKey): A list of Course key object identifying the run whose data we want.
         user (User): The user to authenticate as when making requests to the catalog service.
 
     Returns:
         dict, empty if no data could be retrieved.
     """
-    catalog_integration = CatalogIntegration.current()
-
-    if catalog_integration.enabled:
-        api = create_catalog_api_client(user, catalog_integration)
-
-        data = get_edx_api_data(
-            catalog_integration,
-            user,
-            'course_runs',
-            resource_id=unicode(course_key),
-            cache_key=catalog_integration.CACHE_KEY if catalog_integration.is_cache_enabled else None,
-            api=api,
-            querystring={'exclude_utm': 1},
-        )
-
-        return data if data else {}
-    else:
-        return {}
-
-
-def get_run_marketing_url(course_key, user):
-    """Get a course run's marketing URL from the course catalog service.
+    course_catalog_data_dict = cache.get_many(course_keys)
+    if len(course_catalog_data_dict.keys()) != len(course_keys):
+        found_keys = course_catalog_data_dict.keys()
+        for key in course_keys:
+            if key in found_keys:
+                course_keys.remove(key)
+        log.debug("Catalog data not found in cache against Course Keys: '{}'".format(",".join(course_keys)))
+
+        catalog_integration = CatalogIntegration.current()
+        if catalog_integration.enabled:
+            api = course_discovery_api_client(user)
+
+            catalog_data = get_edx_api_data(
+                catalog_integration,
+                user,
+                'course_runs',
+                api=api,
+                querystring={'keys': ",".join(course_keys), 'exclude_utm': 1},
+            )
+            if catalog_data:
+                for catalog_course_run in catalog_data:
+                    cache.set(catalog_course_run["key"], catalog_course_run, None)
+                    course_catalog_data_dict[catalog_course_run["key"]] = catalog_course_run
+    return course_catalog_data_dict
+
+
+def get_run_marketing_urls(user, course_keys=[]):  # pylint: disable=dangerous-default-value
```

Comment:
@wajeeha-khalid mutable defaults are dangerous. Pylint's trying to tell you that. Don't do this; consider defaulting to None instead.

Also, this module has changed significantly since you opened this PR. You're re-introducing methods that have been removed because they're no longer necessary.